application/dash_application/utility/df_manipulation.py

```diff
@@ -158,19 +158,37 @@ def normalized_merged(df: DataFrame, merged_cols: List[str]):
 _pinery_samples = _pinery_samples.astype({
     PINERY_COL.SequencerRunName: 'str',
     PINERY_COL.LaneNumber: 'int64',
-    "IUSTag": 'str',
+    PINERY_COL.IUSTag: 'str',
     PINERY_COL.GroupID: 'str'})
 # Fill in the "Sample Type" column (Tumor/Reference/Blood/Unknown)
 _pinery_samples[sample_type_col] = _pinery_samples.apply(label_sample_type, axis=1)
 # Drop columns we definitely don't care about.
-_pinery_samples = _pinery_samples.drop(axis=1, columns=[PINERY_COL.NanodropConcentration, PINERY_COL.QubitConcentration, "RunIDandPosition", "TargetedResequencing", "TubeID", "RIN",
-"PrepKit", "PoolName", "Organism", "STRResult", "QubitConcentration", "SampleProvenanceID",
-"TissuePreparation", "TissueType", "SampleName", "RootSampleName", "Version", "NanodropConcentration",
-"TissueOrigin", "Purpose", "SequencerRunName", "SequencingParameters", "UMIs", "GroupIDDescription",
-"DV200", "CreateDate", "InstrumentName", "ParentSampleName", "SubProject", "TemplateType",
-"RunBaseMask", "RunDir", "ExternalName", "Institute", "StudyTitle", "LastModified",
-"GroupID", "SequencerRunPlatformModel", "ReceiveDate", "TissueRegion", "WorkflowType", "Skip",
-"LaneNumber", "LibrarySourceTemplateType"])
+_pinery_samples = _pinery_samples.drop(axis=1, columns=[
+    PINERY_COL.NanodropConcentration,
+    PINERY_COL.QubitConcentration,
+    PINERY_COL.RunIDandPosition,
+    PINERY_COL.TubeID,
+    PINERY_COL.PoolName,
+    PINERY_COL.STRResult,
+    PINERY_COL.QubitConcentration,
+    PINERY_COL.SampleProvenanceID,
+    PINERY_COL.Version,
+    PINERY_COL.NanodropConcentration,
+    PINERY_COL.Purpose,
+    PINERY_COL.SequencingParameters,
+    PINERY_COL.GroupIDDescription,
+    PINERY_COL.CreateDate,
+    PINERY_COL.ParentSampleName,
+    PINERY_COL.TemplateType,
+    PINERY_COL.RunBaseMask,
+    PINERY_COL.RunDir,
+    PINERY_COL.LastModified,
+    PINERY_COL.SequencerRunPlatformModel,
+    PINERY_COL.ReceiveDate,
+    PINERY_COL.TissueRegion,
+    PINERY_COL.WorkflowType,
+    PINERY_COL.Skip,
```

Comment:
44 columns named previously, 24 here - do we no longer need to filter those columns out?

Reply:
Yeah, I could not quite understand why the original version worked. It is removing mandatory fields (`TissuePreparation`, `SampleName`), which should have caused a crashed Dashi. `drop` would have raised an exception if any of the columns to be dropped were missing (column names were different), so they were definitely present.

The way I build my list is to start out with all columns from the previous list and slowly delete the ones that cause Dashi to crash.

Reply:
The bug in ETL caused it to inject columns with NaNs.

Reply:
The remaining columns after the `drop` call:
```
['studyTitle', 'rootSampleName', 'sampleName', 'sequencerRunName', 'laneNumber', 'iusTag', 'instrument_name', 'geo_external_name', 'geo_library_source_template_type', 'geo_organism', 'geo_prep_kit', 'geo_tissue_origin', 'geo_tissue_type', 'umis', 'subproject', 'geo_targeted_resequencing', 'geo_tissue_preparation', 'rin', 'institute', 'geo_group_id', 'dv200', 'sequencing_control_type', 'Sample Type']
```